sherpa/astro/xspec/tests/test_xspec_unit.py

```diff
@@ -1378,3 +1378,265 @@ def test_xsparameter_limits(base):
 
     assert p.max == pytest.approx(8)
     assert p.hard_max == pytest.approx(8)
+
+
+@requires_xspec
+@pytest.mark.parametrize("clsname", ["XSpowerlaw", "XSphabs"])
+def test_model_can_send_spectrumnumber_indiv(clsname):
+    """Check we can send spectrumNumber to individual models.
+
+    All we do is check that the model can be called, and just for
+    additive/multiplicative models that are not expected to use
+    the spectrumNumber argument.
+    """
+
+    from sherpa.astro import xspec
```

Comment:
I recently learned about [pytest's `importorskip`](https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest-importorskip). I'm not suggesting to change the code, I'm just noticing right now (maybe it's been obvious for you for a long time), that that could replace some of our own requires_X decorators if we aver want to. Not saying that we should, as they work fine, I'm just noticing.

Reply:
Extending this detour, I have, in another commit somewhere (for the life of me I don't know in what PR), a fixture that does the 'from sherpa.astro import xpec' line for you (and returns the model component you asked for, so it may or may not be relevant here). That could probably benefit from `importorskip`, as could probably the `requires_xxx` routines (although I'm not sure the semantics actually matches this routine, would need to think about, as our requires_xxx stuff allows for a choice of modules, but this is getting far away into the weeds).

I'm going to resolve this as it's basically to say "don't forget importorskip".